packages/next/src/server/future/route-modules/app-route/module.ts

```diff
@@ -0,0 +1,404 @@
+} from '../route-module'
+import {
+  RequestAsyncStorageWrapper,
+  type RequestContext,
+} from '../../../async-storage/request-async-storage-wrapper'
+import {
+  StaticGenerationAsyncStorageWrapper,
+  type StaticGenerationContext,
+} from '../../../async-storage/static-generation-async-storage-wrapper'
+import {
+  handleBadRequestResponse,
+  handleInternalServerErrorResponse,
+} from '../helpers/response-handlers'
+import { HTTP_METHOD, HTTP_METHODS, isHTTPMethod } from '../../../web/http'
+import { patchFetch } from '../../../lib/patch-fetch'
+import { getTracer } from '../../../lib/trace/tracer'
+import { AppRouteRouteHandlersSpan } from '../../../lib/trace/constants'
+import { getPathnameFromAbsolutePath } from './helpers/get-pathname-from-absolute-path'
+import { proxyRequest } from './helpers/proxy-request'
+import { resolveHandlerError } from './helpers/resolve-handler-error'
+import { wrapRequest } from './helpers/wrap-request'
+import { RouteKind } from '../../route-kind'
+import * as Log from '../../../../build/output/log'
+import { autoImplementMethods } from './helpers/auto-implement-methods'
+import { getNonStaticMethods } from './helpers/get-non-static-methods'
+
+/**
+ * AppRouteRouteHandlerContext is the context that is passed to the route
+ * handler for app routes.
+ */
+export interface AppRouteRouteHandlerContext extends RouteModuleHandleContext {
+  staticGenerationContext: StaticGenerationContext['renderOpts']
+}
+
+/**
+ * AppRouteHandlerFnContext is the context that is passed to the handler as the
+ * second argument.
+ */
+interface AppRouteHandlerFnContext {
+  params?: Params
+}
+
+/**
+ * Handler function for app routes.
+ */
+export type AppRouteHandlerFn = (
+  /**
+   * Incoming request object.
+   */
+  req: Request,
+  /**
+   * Context properties on the request (including the parameters if this was a
+   * dynamic route).
+   */
+  ctx: AppRouteHandlerFnContext
+) => Response
+
+/**
+ * AppRouteHandlers describes the handlers for app routes that is provided by
+ * the userland module.
+ */
+export type AppRouteHandlers = {
+  [method in HTTP_METHOD]?: AppRouteHandlerFn
+}
+
+/**
+ * AppRouteUserlandModule is the userland module that is provided for app
+ * routes. This contains all the user generated code.
+ */
+export type AppRouteUserlandModule = AppRouteHandlers &
+  Pick<AppConfig, 'dynamic' | 'revalidate' | 'dynamicParams' | 'fetchCache'> & {
+    // TODO: (wyattjoh) create a type for this
+    generateStaticParams?: any
+  }
+
+/**
+ * AppRouteRouteModuleOptions is the options that are passed to the app route
+ * module from the bundled code.
+ */
+export interface AppRouteRouteModuleOptions
+  extends RouteModuleOptions<AppRouteUserlandModule> {
+  readonly pathname: string
+  readonly resolvedPagePath: string
+  readonly nextConfigOutput: NextConfig['output']
+}
+
+/**
+ * AppRouteRouteHandler is the handler for app routes.
+ */
+export class AppRouteRouteModule extends RouteModule<
```

Comment:
This is the old `AppRouteRouteHandler`, but now it extends the abstract class `RouteModule` that gets the rest of the shared dependancies.